overlay/overlay_exe/overlay_exe.cpp

```diff
@@ -0,0 +1,175 @@
+     this list of conditions and the following disclaimer in the documentation
+     and/or other materials provided with the distribution.
+   - Neither the name of the Mumble Developers nor the names of its
+     contributors may be used to endorse or promote products derived from this
+     software without specific prior written permission.
+
+   THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+   ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+   LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+   A PARTICULAR PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE FOUNDATION OR
+   CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+   EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+   PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+   PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+   LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+   NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+   SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+#include <windows.h>
+#include <shlwapi.h>
+#include <stdio.h>
+
+#include <string>
+
+#include "../overlay.h"
+
+#define UNUSED(x) ((void)x)
+
+typedef int (*OverlayHelperProcessMain)(unsigned int magic);
+
+// Signal to the overlay DLL that it should not inject
+// into this process.
+extern "C" __declspec(dllexport) void mumbleSelfDetection() {};
+
+// Alert shows a fatal error dialog and waits for the user to click OK.
+static void Alert(LPCWSTR title, LPCWSTR msg) {
+	MessageBox(NULL, msg, title, MB_OK|MB_ICONERROR);
+}
+
+// GetExecutableDirPath returns the directory that
+// mumble.exe resides in.
+static std::wstring GetExecutableDirPath() {
+	wchar_t path[MAX_PATH];
+
+	if (GetModuleFileNameW(NULL, path, MAX_PATH) == 0)
+		return std::wstring();
+
+	if (!PathRemoveFileSpecW(path))
+		return std::wstring();
+
+	std::wstring exe_path(path);
+	return exe_path.append(L"\\");
+}
+
+// ConfigureEnvironment prepares mumble.exe's environment to
+// run mumble_app.dll.
+static bool ConfigureEnvironment() {
+	std::wstring exe_path = GetExecutableDirPath();
+
+	// Remove the current directory from the DLL search path.
+	if (!SetDllDirectoryW(L""))
+		return false;
+
+	// Set mumble.exe's directory as the current working directory.
+	if (!SetCurrentDirectoryW(exe_path.c_str()))
+		return false;
+
+	return true;
+}
+
+// GetAbsoluteMumbleOverlayDllPath returns the absolute path to
+// mumble_ol.dll - the DLL containing the Mumble overlay code.
+static std::wstring GetAbsoluteMumbleOverlayDllPath() {
+	std::wstring exe_path = GetExecutableDirPath();
+	if (exe_path.empty())
+		return std::wstring();
+
+	std::wstring abs_dll_path(exe_path);
+	abs_dll_path.append(L"\\");
+	abs_dll_path.append(L"mumble_ol.dll");
+	return abs_dll_path;
+}
+
+int main(int argc, char **argv) {
+	UNUSED(argc);
+	UNUSED(argv);
```

Comment:
Those are just for preventing the unused warnings, right?

I would prefer

    int main(int /*argc*/, char **/*argv*/) {